src/storage/config_manager.rs

```diff
@@ -0,0 +1,126 @@
+        scheduler: TxnScheduler<E, L>,
+    ) -> Self {
+        StorageConfigManger {
+            kvdb,
+            shared_block_cache,
+            ttl_checker_scheduler,
+            flow_controller,
+            scheduler,
+        }
+    }
+}
+
+impl<EK: Engine, L: LockManager> ConfigManager for StorageConfigManger<EK, L> {
+    fn dispatch(&mut self, mut change: ConfigChange) -> CfgResult<()> {
+        if let Some(ConfigValue::Module(mut block_cache)) = change.remove("block_cache") {
+            if !self.shared_block_cache {
+                return Err("shared block cache is disabled".into());
+            }
+            if let Some(size) = block_cache.remove("capacity") {
+                if size != ConfigValue::None {
+                    let s: ReadableSize = size.into();
+                    // Hack: since all CFs in both kvdb and raftdb share a block cache, we can change
+                    // the size through any of them. Here we change it through default CF in kvdb.
+                    // A better way to do it is to hold the cache reference somewhere, and use it to
+                    // change cache size.
+                    let opt = self.kvdb.get_options_cf(CF_DEFAULT).unwrap(); // FIXME unwrap
+                    opt.set_block_cache_capacity(s.0)?;
+                    // Write config to metric
+                    CONFIG_ROCKSDB_GAUGE
+                        .with_label_values(&[CF_DEFAULT, "block_cache_size"])
+                        .set(s.0 as f64);
+                }
+            }
+        } else if let Some(v) = change.remove("ttl_check_poll_interval") {
+            let interval: ReadableDuration = v.into();
+            self.ttl_checker_scheduler
+                .schedule(TTLCheckerTask::UpdatePollInterval(interval.into()))
+                .unwrap();
+        } else if let Some(ConfigValue::Module(mut flow_control)) = change.remove("flow_control") {
+            if let Some(v) = flow_control.remove("enable") {
+                let enable: bool = v.into();
+                if enable {
+                    for cf in self.kvdb.cf_names() {
+                        self.kvdb
+                            .set_options_cf(cf, &[("disable_write_stall", "true")])
+                            .unwrap();
+                    }
+                    self.flow_controller.enable(true);
+                } else {
+                    for cf in self.kvdb.cf_names() {
+                        self.kvdb
+                            .set_options_cf(cf, &[("disable_write_stall", "false")])
+                            .unwrap();
+                    }
+                    self.flow_controller.enable(false);
+                }
+            }
+        } else if let Some(v) = change.get("scheduler_worker_pool_size") {
+            let pool_size: usize = v.into();
+            let max_pool_size = std::cmp::max(1, SysQuota::cpu_cores_quota() as usize);
+            if pool_size == 0 || pool_size > max_pool_size {
```

Comment:
When a configuration file is loaded or updated, configuration verification is performed and the validate function of each configuration block is called, so I think we'd better move this to validate too.